File: 94-binary-tree-inorder-traversal/94-binary-tree-inorder-traversal.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<int> inorderTraversal(TreeNode* root) {
        vector<int> result, temp;
        
        if(root==nullptr) return result;
        temp = inorderTraversal(root->left);
        result.insert(result.end(), temp.begin(), temp.end());
        
        result.push_back(root->val);
        
        temp = inorderTraversal(root->right);
        result.insert(result.end(), temp.begin(), temp.end());
        
        return result;
    }
};

File: 94-binary-tree-inorder-traversal/94-binary-tree-inorder-traversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def inorderTraversal(self, root: Optional[TreeNode]) -> List[int]:
        if not root:
            return []
        
        res=[]
        
        if root.left:
            res.extend(self.inorderTraversal(root.left))
        
        res.append(root.val)
        
        if root.right:
            res.extend(self.inorderTraversal(root.right))
        
        return res